### DISCUSSION TITLE
config question

### DISCUSSION THREAD

**Discussion by gjblajian** [fastify/fastify#4649] [Q&A] [answered]
Hi guys, I am trying to figure out how to properly use the requestConfig, I am trying to add liveQueryStore to the fastifyContext using the code

app.register(callbackRoutes, { prefix: 'callbacks', config: { liveQueryStore: liveQueryStore } })

the problem is when I am trying to access it VSCode is giving me the deprecation warning

req.context.config.liveQueryStore
on config. The documentation looks like it is telling us to use that path to get the additional config but the deprecation indicates that it should look more like 

req.routeConfig.liveQueryStore

[https://github.com/fastify/fastify/blob/main/docs/Reference/Routes.md#config] One would assume that the documentation is not updated but the github issue [https://github.com/fastify/fastify/pull/4470] indicates that the documentation SHOULD have been updated at the same time as the deprecation warning being added. I just want to know what the right way is now to access my config variable 

and how do I tell typescript the structure of my config object?
so that it knows that the liveQueryStore SHOULD be there

**Comment by metcoder95** [comment, marked as answer]
Hey! Thanks for the heads up!

Quickly answering your question, yes, you should be using `req.routeConfig` instead of `route.config`, as it will be removed in a future major version.

To reflect your requirements in TS, you need to use the `interface augmentation` capabilities that Fastify provides. See the following example: https://github.com/metcoder95/fastify-split-validator/blob/a3ad0a8e32658f559bf9c36509b731bcd6140ef8/index.d.ts#LL10C27-L10C27

Regarding the documentation issue, yeah it is a leftover that we didn't notice. Would you like to submit a PR to adjust the documentation? 🙂
